sled-agent/src/opte/mod.rs

```diff
@@ -4,8 +4,260 @@
 
 //! Interactions with the Oxide Packet Transformation Engine (OPTE)
 
-#[cfg_attr(target_os = "illumos", path = "opte.rs")]
-#[cfg_attr(not(target_os = "illumos"), path = "mock_opte.rs")]
-mod inner;
+use ipnetwork::IpNetwork;
+use macaddr::MacAddr6;
+use slog::Logger;
+use std::net::IpAddr;
+use std::net::Ipv6Addr;
 
-pub use inner::*;
+mod port;
+mod port_manager;
+
+pub use port::Port;
+pub use port_manager::PortManager;
+pub use port_manager::PortTicket;
+
+cfg_if::cfg_if! {
```

Comment:
I'm with you. That was how the implementation prior to this PR worked, with an inner module whose path was chosen based on the platform to get a real or mock implementation. I was hoping that with the changes here, the code would actually require less platform-dependent work, or at least it'd be more contained. That turned out not to be the case, but I didn't go back to that.

Since you also feel it's annoying and risky to have the impl distributed like that, I think I'll spend the time to rework it into a real and dummy impl again. Shouldn't be too much trouble, and it'll ease maintenance burden.

As for the second question, I'm really just trying to simulate the interface, which is basically just argument parsing at this point. I'm not sure I grok your suggestion, though. Could you elaborate? The current implementation will basically have a `PortManager` with `Port` objects for each instance, but those'll just be POD, there's nothing on the system they reflect in any way. I assume that's different from your suggestion, but I'm not clear how so.

Reply:
FYI, simplifying the module organization for different platforms is done in [15668b1](https://github.com/oxidecomputer/omicron/pull/1385/commits/15668b153f6d376a3a0a21d45fa81c54998659ab) and [4edec92](https://github.com/oxidecomputer/omicron/pull/1385/commits/4edec92758c3a02473e58d5f4faed07b3a3cb3b5)

Reply:
I like the change you made - nice!

My suggestion was basically just musing about "how we have been mocking things" vs "how we could be mocking things". We basically use the same code for instances on illumos / non-illumos, and mock out modules at a lower level when very specific invocations are made (e.g., calls to opte, dladm, zones, ipadm, etc). I was wondering, "what about mocking things out at a higher level?", as in, instead of mocking each resource, mocking out the whole call to "instance_ensure", such that the specific resources used can be tracked, but no "fake setup calls" are otherwise made.

I dunno. This is obviously not a blocker for this PR, but something to consider as we use more resources, and the management of those resources gets more complicated. Maybe it's not worth it, as moving up "where we mock stuff" would also prevent a lot of unit tests being written on generic platforms.